### PR TITLE
Enable building with GCC 10.3 for JDK18 and above on Linux (Excluding OpenJ9)

### PR DESCRIPTION
**Pull request by sxa**
Can be implemented now that https://github.com/adoptium/infrastructure/pull/2420 has been merged
Fixes https://github.com/adoptium/temurin-build/issues/2787 (Although for now not including JDK17)

### CODE REVIEW

File: build-farm/platform-specific-configurations/linux.sh [hunk 1]
```diff
@@ -181,16 +181,31 @@ if [ $executedJavaVersion -ne 0 ]; then
...
+    export LD_LIBRARY_PATH=/usr/local/gcc10/lib64:/usr/local/gcc10/lib
+  elif [ -r /usr/bin/gcc-10 ]; then
+    [ -r /usr/bin/gcc-10 ] && export  CC=/usr/bin/gcc-10
+    [ -r /usr/bin/g++-10 ] && export CXX=/usr/bin/g++-10
+  fi
+# Continue to use GCC 7 if present for JDK<=17
 elif [ -r /usr/local/gcc/bin/gcc-7.5 ]; then
   export PATH=/usr/local/gcc/bin:$PATH
-  [ -r /usr/local/gcc/bin/gcc-7.5 ] && export CC=/usr/local/gcc/bin/gcc-7.5
+  [ -r /usr/local/gcc/bin/gcc-7.5 ] && export  CC=/usr/local/gcc/bin/gcc-7.5
```
Comment (karianna) [inline build-farm/platform-specific-configurations/linux.sh:204]:
```suggestion
  [ -r /usr/local/gcc/bin/gcc-7.5 ] && export CC=/usr/local/gcc/bin/gcc-7.5
```

File: build-farm/platform-specific-configurations/linux.sh [hunk 2]
```diff
@@ -181,16 +181,31 @@ if [ $executedJavaVersion -ne 0 ]; then
...
   [ -r /usr/local/gcc/bin/g++-7.5 ] && export CXX=/usr/local/gcc/bin/g++-7.5
   export LD_LIBRARY_PATH=/usr/local/gcc/lib64:/usr/local/gcc/lib
 elif [ -r /usr/bin/gcc-7 ]; then
-  [ -r /usr/bin/gcc-7 ] && export CC=/usr/bin/gcc-7
+  [ -r /usr/bin/gcc-7 ] && export  CC=/usr/bin/gcc-7
```
Comment (karianna) [inline build-farm/platform-specific-configurations/linux.sh:208]:
```suggestion
  [ -r /usr/bin/gcc-7 ] && export CC=/usr/bin/gcc-7
```

**Review by karianna — CHANGES_REQUESTED** [review]
Couple of small whitespace suggestions.  Failed GH Actions will need checking as well.

File: build-farm/platform-specific-configurations/linux.sh (hunk 2 above)
Comment (sxa) [reply build-farm/platform-specific-configurations/linux.sh:208]: I'm not sure I agree with that change - it did it deliberatily to line up `gcc-7` and `g++-7` vertically which I think looks nicer

Comment (karianna) [reply build-farm/platform-specific-configurations/linux.sh:208]: Oh that's what you were doing :-)

**Comment by karianna** [comment]
The GH Actions failing will still need to be addressed and the ECA check?

**Review by andrew-m-leonard — APPROVED** [review]

**Comment by sxa** [comment]
@karianna Bisheng11 seems to be broken in other PRs so that check failing is probably not a blocker

**Review by karianna — APPROVED** [review]

**Comment by gdams** [comment]
> @gdams any idea why the centos7 one may not have got the update from adoptium/infrastructure#2420

@sxa looks like the playbook updater has been failing in Jenkins for some time: https://ci.adoptopenjdk.net/view/Infra/job/centos7_docker_image_updater/. Weirdly it seems to be failing to push  (I'll do some digging)

**Comment by sxa** [comment]
> > @gdams any idea why the centos7 one may not have got the update from [adoptium/infrastructure#2420](https://github.com/adoptium/infrastructure/pull/2420)
> 
> @sxa looks like the playbook updater has been failing in Jenkins for some time: https://ci.adoptopenjdk.net/view/Infra/job/centos7_docker_image_updater/. Weirdly it seems to be failing to push (I'll do some digging)

Let's move further discussion into https://github.com/adoptium/infrastructure/issues/2432

**Comment by sxa** [comment]
I'm comfortable that this is having the desired effect - merging now.

**Pull request merged**